driver/src/main/scala/org/mongodb/scala/Observer.scala

```diff
@@ -35,7 +35,7 @@ import com.mongodb.async.client.{Observer => JObserver, Subscription => JSubscri
  *
  * @tparam T The type of element signaled.
  */
-trait Observer[T] extends JObserver[T] {
+trait Observer[T] extends Subscriber[T] {
```

Comment:
`Observer[T`] extends `Subscriber[T]` the only difference is it can take either a reactive streams `Subscription` or a scala driver `Subscription`.

Reply:
In this case it needs to be able to take either due to the slightly different api of Subscription between the original implementation and reactive streams.